### PR TITLE
HW1 WhackAMole

### PR DESCRIPTION
**Pull request by noemiguzo**
I tried to use Scanner to play, work but not passed  gradle

### CODE REVIEW

File: src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+     */
+    public void updateForScoreMolesLeft() {
+        score++;
+        molesLeft--;
+    }
+
+    /**
+     * Methods that print the grid.
+     */
+    public void printGrid() {
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java]: In order to test printGrid method.
This should return a String.

File: src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+            }
+            grid.append('\n');
+        }
+        System.out.print(grid.toString());
+    }
+
+    /**
+     * Methods that print the grid to user.
+     */
+    public void printGridToUser() {
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java]: In order to test printGridToUser method.
This should return a String.

File: src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+            return;
+        }
+        userGrid.append(String.valueOf(typeCharacter).concat(SPACES));
+    }
+
+    /**
+     * This public main methods.
+     * @param args string
+     */
+    public static void main(final String[] args) {
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java]: Since EDX doesn't require this.
You can remove it.

**Comment by noemiguzo** [comment]
image

**Comment by noemiguzo** [comment]
image

**Comment by noemiguzo** [comment]
image

File: src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+    /**
+     * Methods that print the grid.
+     * @return grid string.
+     */
+    public String printGrid() {
+        StringBuilder grid = new StringBuilder();
+        for (char[] row : moleGrid) {
+            for (char value : row) {
+                grid.append(String.valueOf(value).concat(SPACES));
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java]: String.valueOf is redundant.
concat method is redundant.
Take advantage of StringBuilder

File: src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+     * This methods to verify the moles in the game.
+     * @param userGrid the userGrid.
+     * @param typeCharacter the caracter.
+     */
+    private void existTheMole(final StringBuilder userGrid, char typeCharacter) {
+        if (typeCharacter == MOLE) {
+            userGrid.append(String.valueOf("*").concat(" "));
+            return;
+        }
+        userGrid.append(String.valueOf(typeCharacter).concat(SPACES));
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java]: Same as above

File: src/test/java/org/minions/devfund/noemiguzman/WhackAMoleTest.java [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * It is in charged to define the unit tests for WhackAMole class.
+ */
+public class WhackAMoleTest {
+    public static final int NUM_ATTEMPTS =  10;
+    public static final int GRID_DIMENSION = 10;
```
Comment (carledriss) [inline src/test/java/org/minions/devfund/noemiguzman/WhackAMoleTest.java]: Those constants should be private

File: src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+
+
+    /**
+     * This methods to verify the moles in the game.
+     * @param userGrid the userGrid.
+     * @param typeCharacter the caracter.
+     */
+    private void existTheMole(final StringBuilder userGrid, char typeCharacter) {
+        if (typeCharacter == MOLE) {
+            userGrid.append(String.valueOf("*").concat(" "));
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java]: Take advantage of StringBuilder here.

File: src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+            return;
+        }
+        userGrid.append(String.valueOf(typeCharacter).concat(SPACES));
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java]: Same as above

File: src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+        return attemptsLeft;
+    }
+
+
+    /**
+     * This methods to verify the moles in the game.
+     * @param userGrid the userGrid.
+     * @param typeCharacter the caracter.
+     */
+    private void existTheMole(final StringBuilder userGrid, char typeCharacter) {
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/noemiguzman/WhackAMole.java]: Is it really needed this method ?

**Comment by carledriss** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**